materializationengine/admin.py

```diff
@@ -1,10 +1,105 @@
-from flask_admin import Admin
+from flask_admin import Admin, AdminIndexView, BaseView, expose
 from flask_admin.contrib.sqla import ModelView
 from dynamicannotationdb.models import AnalysisVersion, AnalysisTable
+from flask import request, current_app, g, render_template
+from middle_auth_client import auth_required
+from dynamicannotationdb.migration import DynamicMigration, run_alembic_migration
+from sqlalchemy.engine.url import make_url
+from materializationengine.info_client import (
+    get_datastacks,
+    get_relevant_datastack_info,
+)
+
+
+class MatAdminIndexView(AdminIndexView):
+    @expose("/", methods=["GET"])
+    @auth_required
+    def index(self):
+        return super(MatAdminIndexView, self).index()
+
+    def is_accessible(self):
+        @auth_required
+        def helper():
+            return True
+
+        return helper()
 
 
 def setup_admin(app, db):
-    admin = Admin(app, name="materializationengine")
+    admin = Admin(app, name="materializationengine", index_view=MatAdminIndexView())
+    admin.add_view(MigrationView(name="Migration"))
     admin.add_view(ModelView(AnalysisVersion, db.session))
     admin.add_view(ModelView(AnalysisTable, db.session))
     return admin
+
+
+def get_allowed_aligned_volumes():
+    with current_app.app_context():
+        datastacks = get_datastacks()
+        aligned_volumes = []
+        for datastack in datastacks:
+            aligned_volume_name, pcg_table_name = get_relevant_datastack_info(datastack)
+            aligned_volumes.append(aligned_volume_name)
+    return aligned_volumes
+
+
+class MigrationView(BaseView):
+    def is_accessible(self):
+        @auth_required
+        def helper():
+            return True
+
+        return helper() and g.get("auth_user", {}).get("admin", False)
+
+    def inaccessible_callback(self, name, **kwargs):
+        return render_template("admin/403.html"), 403
+
+    @expose("/")
+    @auth_required
+    def index(self):
+        aligned_volumes = get_allowed_aligned_volumes()
+        return self.render("admin/migration.html", aligned_volumes=aligned_volumes)
+
+    @expose("/migrate_static_schemas", methods=["POST"])
+    @auth_required
+    def migrate_static_schemas(self):
+        sql_url = request.form.get("sql_url")
+        aligned_volume = request.form.get("aligned_volume")
+        sql_base_uri = sql_url.rpartition("/")[0]
+        sql_uri = make_url(f"{sql_base_uri}/{aligned_volume}")
+        migrator = run_alembic_migration(str(sql_uri))
+        return self.render(
+            "admin/migration.html",
+            message=migrator,
+            aligned_volumes=get_allowed_aligned_volumes(),
+        )
+
+    @expose("/migrate_annotation_schemas", methods=["POST"])
+    @auth_required
```

Comment:
shouldn't this require admin?

Reply:
When I tested it if the user is not an admin it hides the migration tab in the admin view, but we should use the decorator as well, I forgot about the auth requires admin decorator, simple fix.